File: src/test/java/org/dedda/bratwurst/lang/io/file/FileImportTest.java
package org.dedda.bratwurst.lang.io.file;

import org.dedda.bratwurst.FileSystemTestCase;
import org.dedda.bratwurst.lang.BWString;
import org.dedda.bratwurst.lang.BWVariable;
import org.dedda.bratwurst.lang.scope.Scope;
import org.testng.annotations.Test;

import java.io.File;

import static org.testng.Assert.assertEquals;

/**
 * Created by dedda on 1/28/16.
 *
 * @author dedda
 */
public class FileImportTest extends FileSystemTestCase {

    @Test
    public void testRun() throws Exception {
        String filename = genericFileName();
        String content = "Some content to test";
        File file = new File(filename);
        createFile(file, content);
        Scope scope = createEmptyScope();
        scope.setVariable(new BWVariable("var", new BWString(filename)));
        FileImport fileImport = new FileImport(0, "var");
        fileImport.run(scope);
        BWVariable var = scope.getVariable("var");
        BWString bwString = (BWString) var.getValue();
        assertEquals(content, bwString.getStringValue());
        removeFile(file);
    }
}

File: src/test/java/org/dedda/bratwurst/lang/io/file/FileImportTest.kt
package org.dedda.bratwurst.lang.io.file

import org.dedda.bratwurst.FileSystemTestCase
import org.dedda.bratwurst.lang.BWString
import org.dedda.bratwurst.lang.BWVariable
import org.testng.Assert
import org.testng.annotations.Test
import java.io.File

/**
 * Created by dedda on 1/28/16.
 *
 * @author dedda
 */
class FileImportTest : FileSystemTestCase() {
    @Test
    @Throws(Exception::class)
    fun testRun() {
        val filename = genericFileName()
        val content = "Some content to test"
        val file = File(filename)
        createFile(file, content)
        val scope = createEmptyScope()
        scope.setVariable(BWVariable("var", BWString(filename)))
        val fileImport = FileImport(0, "var")
        fileImport.run(scope)
        val `var` = scope.getVariable("var")
        val bwString = `var`.value as BWString
        Assert.assertEquals(content, bwString.stringValue)
        removeFile(file)
    }
}